src/ssl_stubs.c

```diff
@@ -1751,8 +1751,11 @@ CAMLprim value ocaml_ssl_flush(value socket)
   bio = SSL_get_wbio(ssl);
   if(bio)
   {
-    /* TODO: raise an error */
-    assert(BIO_flush(bio) == 1);
+    int ret = BIO_flush(bio);
+    if (ret != 1) {
```

Comment:
I think this needs to acquire the runtime lock again before raising.

Reply:
done too.